docs/cart-and-checkout-blocks/additional-checkout-fields.md

```diff
@@ -569,6 +566,426 @@ add_action(
+
+### JSON Schema Structure
+
+Each schema in the array should be a valid JSON Schema object that defines conditions for when the property should be applied. The schema is evaluated against the current cart and checkout state, which includes all field values and various options (payment, shipping, customer).
+
+Basic structure of a JSON Schema object:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "fieldId": {
+      "enum": ["value1", "value2"]
+    }
+  },
+  "required": ["fieldId"]
+}
+```
+
+If you're not familiar with JSON Schema, you can get a quick introduction to it [from the official website](https://json-schema.org/understanding-json-schema/basics), or from one of the libraries used [like AJV](https://ajv.js.org/json-schema.html) or [OPIS.](https://opis.io/json-schema/2.x/examples.html) Checkout builds an abstraction on top of both of them.
+
+### Document object
+
+When you're writing your rules, you're writing a partial schema for the document object, essentially describing the ideal state you want for your field to be required or hidden. An example of the document object looks like this:
+
+<!-- markdownlint-disable MD033 -->
+<details>
+	<summary>Document object</summary>
+
+```json
+{
+	"cart": {
+		"coupons": [
+			"my_coupon"
+		],
+		"shipping_rates": [
+			"free_shipping:1"
+		],
+		"items": [
+			27,
+			27,
+			68
+		],
+		"items_type": [
+			"simple",
+			"variation"
+		],
+		"items_count": 3,
+		"items_weight": 0,
+		"needs_shipping": true,
+		"prefers_collection": false,
+		"totals": {
+			"totalPrice": 6600,
+			"totalTax": 600
+		},
+		"extensions": {}
+	},
+	"checkout": {
+		"create_account": false,
+		"customer_note": "",
+		"additional_fields": {
+			"namespace/mycontact-field": "myvalue"
+		},
+		"payment_method": "bacs"
+	},
+	"customer": {
+		"id": 1,
+		"billing_address": {
+			"first_name": "First Name",
+			"last_name": "Last Name",
+			"company": "Company",
+			"address_1": "Address 1",
+			"address_2": "Address 2",
+			"city": "City",
+			"state": "State",
+			"postcode": "08000",
+			"country": "US",
+			"email": "email@example.com",
+			"phone": "1234567890",
+			"namespace/myfield": "myvalue"
+		},
+		"shipping_address": {
+			"first_name": "First Name",
+			"last_name": "Last Name",
+			"company": "Company",
+			"address_1": "Address 1",
+			"address_2": "Address 2",
+			"city": "City",
+			"state": "State",
+			"postcode": "08000",
+			"country": "US",
+			"phone": "1234567890",
+			"namespace/myfield": "myvalue"
+		},
+		"address": {
+			"first_name": "First Name",
+			"last_name": "Last Name",
+			"company": "Company",
+			"address_1": "Address 1",
+			"address_2": "Address 2",
+			"city": "City",
+			"state": "State",
+			"postcode": "08000",
+			"country": "US",
+			"phone": "1234567890",
+			"namespace/myfield": "myvalue"
+		}
+	}
+}
+```
+
+</details>
+<!-- markdownlint-enable MD033 -->
+
+
+It's full schema is this one:
```

Comment:
How can we ensure we keep this in sync with the actual code?

Reply:
no idea yet, markdown doesn't support pulling up from a separate source, maybe switch to MDX will help here.